examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2218 @@
+
+  template <int dim>
+  class Discretization : public dealii::ParameterAcceptor
+  {
+  public:
+    Discretization(const MPI_Comm &     mpi_communicator,
+                   dealii::TimerOutput &computing_timer,
+                   const std::string &  subsection = "Discretization");
+
+    void setup();
+
+    const MPI_Comm &mpi_communicator;
+
+    dealii::parallel::distributed::Triangulation<dim> triangulation;
+
+    const dealii::MappingQ<dim>   mapping;
+    const dealii::FE_Q<dim>       finite_element;
+    const dealii::QGauss<dim>     quadrature;
+    const dealii::QGauss<dim - 1> face_quadrature;
+
+  private:
+    dealii::TimerOutput &computing_timer;
+
+    double immersed_disc_length;
+    double immersed_disc_height;
+    double immersed_disc_object_position;
+    double immersed_disc_object_diameter;
+
+    unsigned int refinement;
+  };
+
+  // @sect4{Declaration of <code>OfflineData</code> class template}
+
+  // The class OfflineData is initializes (and "owns")
+  // pretty much all the components of the discretization that
+  // do not evolve in time. In particular: dof_handler, sparsity
+  // patterns, boundary maps, lumped mass matrix, and other matrices
+  // and vectors that do not change in time are members of this class.
+  // The term "offline" here refers to the idea that all the class members
+  // of <code>OfflineData</code> are initialized and assigned values
+  // a "time step zero" and are not meant to be modified at any other later
+  // time step. For instance, the sparsity pattern should not
+  // change as we advance in time (we are not doing any form of adaptivity in
+  // space). Similarly, the entries of the vector
+  // <code>lumped_mass_matrix</code> should not be modified as we advance in
+  // time either.
+  //
+  // Placeholder: Say something about BoundaryNormalMap.
+
+  template <int dim>
+  class OfflineData : public dealii::ParameterAcceptor
+  {
+  public:
+    using BoundaryNormalMap =
+      std::map<dealii::types::global_dof_index,
+               std::tuple<dealii::Tensor<1, dim> /* normal */,
+                          dealii::types::boundary_id /* id */,
+                          dealii::Point<dim>> /* position */>;
+
+    OfflineData(const MPI_Comm &           mpi_communicator,
+                dealii::TimerOutput &      computing_timer,
+                const Discretization<dim> &discretization,
+                const std::string &        subsection = "OfflineData");
+
+    void setup();
+    void assemble();
+
+    dealii::DoFHandler<dim> dof_handler;
+
+    std::shared_ptr<const dealii::Utilities::MPI::Partitioner> partitioner;
+
+    unsigned int n_locally_owned;
+    unsigned int n_locally_relevant;
+
+    dealii::SparsityPattern sparsity_pattern;
+
+    BoundaryNormalMap boundary_normal_map;
+
+    dealii::SparseMatrix<double>                  lumped_mass_matrix;
+    std::array<dealii::SparseMatrix<double>, dim> cij_matrix;
```

Comment:
I think it would be useful to have comments here explaining what these objects are.